gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -211,7 +211,9 @@ private static StorageDescriptor getStorageDescriptor(HiveRegistrationUnit unit)
     State props = unit.getStorageProps();
     StorageDescriptor sd = new StorageDescriptor();
     sd.setParameters(getParameters(props));
-    sd.setCols(getFieldSchemas(unit));
+    if (unit instanceof HiveTable) {
```

Comment:
I believe this should happen inside `addSchemaProperties` method

Reply:
Due to we skip the schema registration in partition, the conversion from HivePartition to Partition will fail due to below exception. The addSchemaProperties is for HivePartition creation, not the HivePartition to Partition conversion.

org.apache.hadoop.hive.serde2.avro.AvroSerdeException: Neither avro.schema.literal nor avro.schema.url specified, can't determine table schema
        at org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils.determineSchemaOrThrowException(AvroSerdeUtils.java:119)
        at org.apache.hadoop.hive.serde2.avro.AvroSerDe.determineSchemaOrReturnErrorSchema(AvroSerDe.java:177)
        at org.apache.hadoop.hive.serde2.avro.AvroSerDe.initialize(AvroSerDe.java:103)
        at org.apache.hadoop.hive.serde2.avro.AvroSerDe.initialize(AvroSerDe.java:80)
        at org.apache.hadoop.hive.serde2.SerDeUtils.initializeSerDe(SerDeUtils.java:520)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreUtils.getDeserializer(HiveMetaStoreUtils.java:399)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreUtils.getFieldSchemas(HiveMetaStoreUtils.java:356)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreUtils.getStorageDescriptor(HiveMetaStoreUtils.java:214)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreUtils.getPartition(HiveMetaStoreUtils.java:164)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreBasedRegister.addOrAlterPartition(HiveMetaStoreBasedRegister.java:458)
        at org.apache.gobblin.hive.metastore.HiveMetaStoreBasedRegister.registerPath(HiveMetaStoreBasedRegister.java:159)
        at org.apache.gobblin.hive.HiveRegister$1.call(HiveRegister.java:109)
        at org.apache.gobblin.hive.HiveRegister$1.call(HiveRegister.java:93)
        at org.apache.gobblin.util.executors.MDCPropagatingCallable.call(MDCPropagatingCallable.java:42)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Reply:
What I meant was: The determination of HiveRegistrationUnit is an instance of HiveTable or HivePartition (so that we decide to add schema into storage properties or not)should not happen here but in `addSchemaPropertiesIfRequired` only.  The exception you saw if because `getDeserializer `, as the conversion from Gobblin's Hive object to the real Hive object, requires schema related info to be there for SerDe process, which should not be there totally if a converted object is an HivePartition.

Reply:
@autumnust , we already had differentiation in addSchemaPropertiesIfRequired method, where we use `isRegisterSchema` flag to check if a schema needs to be added. The logic here is to convert Gobblin hive obj to native hive obj, which should follow the same check, otherwise we will see the exception due to a missing schema. Please check my latest code change.